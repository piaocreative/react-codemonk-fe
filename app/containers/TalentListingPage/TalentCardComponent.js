import React, { useState } from 'react';
import { H5, P, H3, Badge, Button } from 'components';
import SVG from 'react-inlinesvg';
import get from 'lodash/get';
import {
  teamIcon,
  certificationsIcon,
  industryIcon,
  diamondIcon,
  userProfileIcon,
  timeXZone,
  teamPreferenceArray,
} from 'containers/App/constants';
import { propTypes } from 'containers/proptypes';
import containerMessage from 'containers/messages';
import HireTalent from 'containers/TalentListingPage/HireTalent';
import { UserProfile, UserProfileImg, PreferencesList } from 'containers/MyProfilePage/styles';
import { getArrayLabels, getTimzoneOffest } from 'containers/MyProfilePage/components/utils';
import { TalentCard } from './styles';

export const TalentCardComponent = ({ talentData, project = {} }) => {
  const [hireModal, setHireModal] = useState(false);
  const [hireData, setHireData] = useState({});

  const hireTalent = e => {
    e.preventDefault();
    e.stopPropagation();
    const projectTalentId = get(project, 'projectId', '');
    const projectName = get(project, 'projectName', '');
    const projectSummary = get(project, 'projectDescription', '');
    setHireModal(true);
    setHireData({ projectTalentId, projectName, projectSummary });
  };
  const hireModalClose = () => {
    setHireModal(false);
  };
  const id = get(talentData, '_id', '');
  return (
    <>
      <TalentCard to={`/client/talent-profile/${id}`} target="_blank" key={id} className="card-sm flex-1 d-flex flex-column">
        <div className="d-flex flex-0 justify-content-between align-items-center">
          <UserProfile>
            <UserProfileImg>
              <img
                src={get(talentData, 'profilePicture', '')}
                alt="user profile"
                onError={e => {
                  e.target.onerror = null;
                  e.target.src = userProfileIcon;
                }}
              />
            </UserProfileImg>
          </UserProfile>
          <Button className="btn btn-sm btn-outline" onClick={e => hireTalent(e)}>
            <span>{containerMessage.btnHire.defaultMessage}</span>
          </Button>
        </div>
        <div className="d-flex flex-0 mt-3">
          <div className="d-flex justify-content-between w-100 flex-column">
            <H3>
              {get(talentData, 'firstName')} {get(talentData, 'lastName')}.
            </H3>
            <P className="p14 mb-0" opacityVal="0.5">
              {get(talentData, 'city')}, {get(talentData, 'country')} ({getTimzoneOffest(timeXZone, 'name', get(talentData, 'timeZone'))})
            </P>
          </div>
        </div>
        <hr />
        <div className="flex-0">
          <P className="p14 mb-2" opacityVal="0.5">
            {get(talentData, 'yearsOfExperience')}
          </P>
          <H5>{get(talentData, 'primaryRole')}</H5>
        </div>
        <hr />
        {get(talentData, 'skills') && get(talentData, 'skills', []).length >= 1 ? (
          <>
            <div className="d-flex flex-wrap flex-0">
              {get(talentData, 'skills', [])
                .slice(0, 4)
                .map(subItem => (
                  <Badge key={subItem.name} className="primary badge-sm me-2 mb-2">
                    {subItem.name}
                  </Badge>
                ))}
              {get(talentData, 'skills', []).length > 4 && (
                <Badge className="primary badge-sm me-2 mb-2">+{get(talentData, 'skills', []).length - 4} Skills</Badge>
              )}
            </div>
            <hr />
          </>
        ) : (
          ''
        )}
        <PreferencesList className="d-flex flex-column flex-1">
          {get(talentData, 'certificateDetails') && get(talentData, 'certificateDetails', []).length >= 1 ? (
            <li>
              <SVG src={certificationsIcon} />
              <P className="p16 mb-0 text-truncate d-block" opacityVal="0.5">
                {get(talentData, 'certificateDetails', [])
                  .map(obj => obj.name)
                  .map(i => i)
                  .join(', ')}
              </P>
            </li>
          ) : (
            ''
          )}
          {get(talentData, 'industries') && get(talentData, 'industries', []).length >= 1 ? (
            <li>
              <SVG src={industryIcon} />
              <P className="p16 mb-0 text-truncate d-block" opacityVal="0.5">
                {get(talentData, 'industries', [])
                  .map(i => i)
                  .join(', ')}
              </P>
            </li>
          ) : (
            ''
          )}
          {get(talentData, 'companyCultures') && get(talentData, 'companyCultures', []).length >= 1 ? (
            <li>
              <SVG src={diamondIcon} />
              <P className="p16 mb-0 text-truncate d-block" opacityVal="0.5">
                {get(talentData, 'companyCultures', [])
                  .map(i => i)
                  .join(', ')}
              </P>
            </li>
          ) : (
            ''
          )}
          {get(talentData, 'teamPreference') && get(talentData, 'teamPreference', []).length >= 1 ? (
            <li>
              <SVG src={teamIcon} />
              <P className="p16 mb-0 text-truncate d-block" opacityVal="0.5">
                {getArrayLabels(get(talentData, 'teamPreference', []), teamPreferenceArray)
                  .map(i => i)
                  .join(', ')}
              </P>
            </li>
          ) : (
            ''
          )}
        </PreferencesList>
        <hr />
        <div className="flex-0">
          <P className="p14 mb-2" opacityVal="0.5">
            Formerly at
          </P>
          <P className="p16 mb-0">{get(talentData, 'formerEmployer', [])}</P>
        </div>
      </TalentCard>
      {hireModal && <HireTalent talentId={id} showHireModal={hireModal} hireModalClose={() => hireModalClose()} hireData={hireData} />}
    </>
  );
};

TalentCardComponent.propTypes = propTypes;

export default TalentCardComponent;
