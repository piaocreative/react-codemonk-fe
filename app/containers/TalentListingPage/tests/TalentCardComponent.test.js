/**
 * Test the TalentCardComponent
 */

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { TalentCardComponent as MainForm } from '../TalentCardComponent';

jest.mock('utils/request');
const props = {
  talentData: {
    _id: '5f046da2550f5f00078d7c97',
    profilePicture: 'https://s3-eu-west-1.amazonaws.com/s3-codemonk-static-content/development-proflie-pictures/5f046da2550f5f00078d7c96',
    name: 'harshad parmar',
    teamPreference: ['x-large-team'],
    assignments: ['short-term-onsite'],
    skills: [
      {
        _id: '600ebadd3019e157a2224365',
        name: 'Amazon Redshift',
        rate: 8,
      },
    ],
    certificateDetails: [],
    workExperience: [
      {
        _id: '600ebb063019e157a2224366',
        jobTitle: 'Developer',
        employmentType: 'Contract',
        employer: 'Innovify',
        country: 'Andorra',
        startDate: '2021-01-21T00:00:00.000Z',
        endDate: '2021-01-25T00:00:00.000Z',
        isPresent: true,
        shortDescription: '<p>COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895</p>\n',
      },
    ],
    currency: 'USD',
    city: 'ahmedabad',
    country: 'Algeria',
    timeZone: 'Pacific/Midway',
    experienceOrder: '4',
    primaryRole: 'Product (UI) designer',
    professionalSummary:
      '<p>COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895COD-895</p>\n',
    availability: true,
    workPreference: ['fulltime'],
    firstName: 'harshad',
    lastName: '',
    yearsOfExperience: 'Principal ',
    ratePerHour: 130,
    formerEmployer: 'Innovify',
  },
};

describe('<TalentCardComponent />', () => {
  const renderer = new ShallowRenderer();
  it('should render and match the snapshot', () => {
    renderer.render(<MainForm {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
