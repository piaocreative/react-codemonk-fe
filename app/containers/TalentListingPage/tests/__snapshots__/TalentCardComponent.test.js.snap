// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TalentCardComponent /> should render and match the snapshot 1`] = `
<React.Fragment>
  <ForwardRef(styles__TalentCard)
    className="card-sm flex-1 d-flex flex-column"
    target="_blank"
    to="/client/talent-profile/5f046da2550f5f00078d7c97"
  >
    <div
      className="d-flex flex-0 justify-content-between align-items-center"
    >
      <ForwardRef(styles__UserProfile)>
        <ForwardRef(styles__UserProfileImg)>
          <img
            alt="user profile"
            onError={[Function]}
            src="https://s3-eu-west-1.amazonaws.com/s3-codemonk-static-content/development-proflie-pictures/5f046da2550f5f00078d7c96"
          />
        </ForwardRef(styles__UserProfileImg)>
      </ForwardRef(styles__UserProfile)>
      <Button
        className="btn btn-sm btn-outline"
        onClick={[Function]}
      >
        <span>
          Hire
        </span>
      </Button>
    </div>
    <div
      className="d-flex flex-0 mt-3"
    >
      <div
        className="d-flex justify-content-between w-100 flex-column"
      >
        <ForwardRef(H3)>
          harshad
           
          
          .
        </ForwardRef(H3)>
        <ForwardRef(P)
          className="p14 mb-0"
          opacityVal="0.5"
        >
          ahmedabad
          , 
          Algeria
           (
          GMT-11:00
          )
        </ForwardRef(P)>
      </div>
    </div>
    <hr />
    <div
      className="flex-0"
    >
      <ForwardRef(P)
        className="p14 mb-2"
        opacityVal="0.5"
      >
        Principal 
      </ForwardRef(P)>
      <ForwardRef(H5)>
        Product (UI) designer
      </ForwardRef(H5)>
    </div>
    <hr />
    <React.Fragment>
      <div
        className="d-flex flex-wrap flex-0"
      >
        <ForwardRef(Badge)
          className="primary badge-sm me-2 mb-2"
        >
          Amazon Redshift
        </ForwardRef(Badge)>
      </div>
      <hr />
    </React.Fragment>
    <ForwardRef(styles__PreferencesList)
      className="d-flex flex-column flex-1"
    >
      
      
      
      <li>
        <InlineSVG
          cacheRequests={true}
          src="IMAGE_MOCK"
          uniquifyIDs={false}
        />
        <ForwardRef(P)
          className="p16 mb-0 text-truncate d-block"
          opacityVal="0.5"
        >
          
        </ForwardRef(P)>
      </li>
    </ForwardRef(styles__PreferencesList)>
    <hr />
    <div
      className="flex-0"
    >
      <ForwardRef(P)
        className="p14 mb-2"
        opacityVal="0.5"
      >
        Formerly at
      </ForwardRef(P)>
      <ForwardRef(P)
        className="p16 mb-0"
      >
        Innovify
      </ForwardRef(P)>
    </div>
  </ForwardRef(styles__TalentCard)>
</React.Fragment>
`;
